### DISCUSSION TITLE
yubikey HMAC-SHA1

### DISCUSSION THREAD

**Discussion by Marucins** [Kunzisoft/KeePassDX#1195] [Ideas]
Hi
What about support for unlocking the database using yubikey?

**Comment by J-Jamet** [comment]
https://github.com/Kunzisoft/KeePassDX/issues/8

**Reply by mora-phi** [reply]
Seems it is included in v3.5.0 (dev set to done in various issues opened on the subject).
Is there any way of testing this new version ? I entered beta testing program in Google PlayStore, but my version is still 3.4.5 as of now...

**Reply by J-Jamet** [reply]
The developments of the functionality are finished but still need to be debugged. And version 3.5.0 contains many features. Maybe I will create an intermediate version with less features.

If you want to test the development version, you have to compile the program. Note that the Yubikey functionality needs an Intent to a driver I'm creating, with the same features as yKDroid, but won't work if the intent name request is different.

**Reply by mora-phi** [reply]
Ok thanks, I can wait :-)

A good thing to keep an eye on would be "Quick Unlock" of DB using biometrics, when there is a Yubikey involved.
Indeed, in a competing app (the only one with Yubikey support using yKDroid) the user has to enter the full passphrase and do the yubikey authentication again, very often. Sometimes the quick unlock works, but very often both passphrase and key auth have to be entered from scratch, which is a pain.

**Reply by J-Jamet** [reply]
For database **save**, you will have to activate the Yubikey every time.
Concretely, the workflow remains the same on KeePassDX for unlocking, and if you add a hardware key, it will be simply requested in addition to the usual connection.

**Comment by FreedomIsAChoice** [comment]
Hello,
I am using KeePassDX on my mobile and I want to use a YubiKey 5 NFC (USB-C) for unlocking my database.
Is now the use of a YubiKey fully supported in KeePassDX?
If so, where can I find a support article in the Wiki for configuring a YubiKey correctly?
Thank you in advance for your help!

**Reply by J-Jamet** [reply]
>  Is now the use of a YubiKey fully supported in KeePassDX?

With the beta app Key Driver, yes.

>  If so, where can I find a support article in the Wiki for configuring a YubiKey correctly?

There's still a lot of work to be done. https://github.com/Kunzisoft/KeePassDX/wiki/Hardware-Key#configuration
